server/src/api/v1.py

```diff
@@ -652,3 +656,60 @@ def queue_wait_time_percentiles_get():
             queue["wait_times"]
         )
     return queue_percentile_data
+
+
+def generate_token(permissions, secret_key):
+    """Generates JWT token with queue permission given a secret key"""
+    expiration_time = datetime.utcnow() + timedelta(seconds=2)
+    token_payload = {
+        "exp": expiration_time,
+        "iat": datetime.now(timezone.utc),  # Issued at time
+        "sub": "access_token",
+        "permissions": permissions,
+    }
+    token = jwt.encode(token_payload, secret_key, algorithm="HS256")
+    return token
+
+
+def validate_client_key_pair(client_id: str, client_key: str):
+    """
+    Checks client_id and key pair for validity and returns their permissions
+    """
+    client_key_bytes = client_key.encode("utf-8")
+    client_permissions_entry = database.mongo.db.client_permissions.find_one(
+        {
+            "client_id": client_id,
+        }
+    )
+
+    if client_permissions_entry is None or not bcrypt.checkpw(
+        client_key_bytes,
+        client_permissions_entry["client_secret_hash"].encode("utf8"),
+    ):
+        return None
+    permissions = client_permissions_entry["permissions"]
+    return permissions
+
+
+SECRET_KEY = os.environ.get("JWT_SIGNING_KEY")
+
+
+@v1.post("/oauth2/token")
+def authenticate_client_post():
```

Comment:
Can we rename this to `retrieve_token()`? I think that's a more descriptive name for what we're doing.